sticker/src/depparse/post_processing.rs

```diff
@@ -29,6 +31,48 @@ where
     }
 }
 
+/// Break cycles in the graph.
+///
+/// Panics when a token does not have a head.
+pub fn break_cycles(sent: &mut Sentence, root_idx: usize) {
+    loop {
+        let components = {
+            let digraph: &DiGraph<_, _> = (&*sent).into();
```

Comment:
This looks somewhat ugly, to say the least. Perhaps a `.graph()` method on `Sentence` would make for some nicer looking code? I don't remember if we discussed that in the issues at the `conllx-rs` repo.
```Rust
let graph = sent.graph();
```

or even more concise (untested, don't know if lifetimes are an issue here):
```Rust
let components = tarjan_scc(sent.graph())
                .into_iter()
                .filter(|c| c.len() > 1)
                .collect::<Vec<_>>();
```

Reply:
This is indeed missing in `conllx-rs`. Another annoyance is that `DepGraphMut` does not have a method to remove edges (something like `remove_deprel(dependent: usize)`). I'll add issues.

Reply:
https://github.com/danieldk/conllx-rs/issues/11
https://github.com/danieldk/conllx-rs/issues/10